src/coprocessor/readpool_impl.rs

```diff
@@ -53,15 +56,35 @@ thread_local! {
                 HashMap::default(),
         }
     );
+
+    static TLS_ENGINE_ANY: RefCell<Option<Box<dyn Any>>> = RefCell::new(None);
```

Comment:
Seems we have 2 `TLS_ENGINE_ANY`. I think we only need 1, and it's better to move it to other somewhere, so that we can also avoid pass an `Engine` to `create_raft_storage` and `build_read_pool`.